### PR TITLE
Improve user-friendliness of the `clean` command

### PR DESCRIPTION
**Pull request by cubranic**
- improves the help text and progress messages to make it clearer what's being cleaned means
- have a way to undo a "clean" request
- signal user errors with non-zero exit code

### CODE REVIEW

**Comment by cubranic** [comment]
@jrs65 If you agree with how the new "clean --cancel" feature operates, I will add the same mechanism to "sync".

**Comment by jrs65** [comment]
Oh this is a great feature. Implementation looks good too.

File: alpenhorn/client.py [hunk 1]
```diff
@@ -425,47 +425,72 @@ def verify(node_name, md5, fixdb, acq):
...
 @click.option('--now', '-n', help='Force immediate removal.', is_flag=True)
 @click.option('--target', metavar='TARGET_GROUP', default=None, type=str,
               help='Only clean files already available in this group.')
 @click.option('--acq', metavar='ACQ', default=None, type=str,
               help='Limit removal to acquisition.')
-def clean(node_name, days, force, now, target, acq):
+def clean(node_name, days, cancel, force, now, target, acq):
     """Clean up NODE by marking older files as potentially removable.
 
+    Files will never be removed until they are available on at least two
```
Comment (jrs65) [inline alpenhorn/client.py]: In this documentation you might want to have a longer comment about the effects of the `--cancel` option. Given it completely inverts the operation that is to be done, we might want to discuss a little more.

Comment (cubranic) [reply alpenhorn/client.py]: I was going to and then forgot to do it before committing. I'll add now.

**Pull request merged**